File: DemoMethods/PythonDemo/demo_methods/Basic/BoostingDisabled.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class BoostingDisabled(object):
    @staticmethod
    def boosting_disabled(city="London", country="Denmark"):
        with DocumentStoreHolder.get_store().open_session() as session:
            orders = list(session.query(object_type=Order, index_name="OrderByCompanyAndCountry",
                                        wait_for_non_stale_results=True,
                                        nested_object_types={"OrderedAt": datetime, "RequireAt": datetime,
                                                             "ShippedAt": datetime, "ShipTo": AddressC,
                                                             "Lines": OrderLine}).
                          where(ShipTo_City=city, ShipTo_Country=country))
            
            return orders

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/basic/BoostingDisabled.java
package net.ravendb.demo.basic;

import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Order;
import net.ravendb.demo.indexes.OrderByCompanyAndCountry;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
public class BoostingDisabled {

    @GetMapping("/basic/boostingDisabled")
    public List<Order> boostingDisabled(
            @RequestParam(value = "city", defaultValue = "London") String city,
            @RequestParam(value = "country", defaultValue = "Denmark") String country) {

        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            List<Order> orders = session.query(Order.class, OrderByCompanyAndCountry.class)
                    .whereEquals("ShipTo_City", city)
                    .orElse()
                    .whereEquals("ShipTo_Country", country)
                    .toList();

            return orders;
        }
    }
}
